apollo-router-core/src/spec/query.rs

```diff
@@ -45,29 +58,24 @@ impl Query {
         }
     }
 
-    pub fn parse(query: impl Into<String>, schema: &Schema) -> Option<Self> {
+    #[tracing::instrument(skip_all, level = "trace")]
+    pub fn parse(query: impl Into<String>) -> Option<Self> {
         let string = query.into();
-        let mut fragments = schema
-            .fragments()
-            .map(|(key, value)| (key.clone(), value.clone()))
-            .collect::<HashMap<String, Vec<_>>>();
 
         let parser = apollo_parser::Parser::new(string.as_str());
         let tree = parser.parse();
+        let errors = tree
+            .errors()
+            .map(|err| format!("{:?}", err))
+            .collect::<Vec<_>>();
```

Comment:
nit: maybe do the let errors = tree.errors() step here, and the map / collect in the if ?

Reply:
it's actually more optimal this way as the iterator is created only once and the vector does not allocate if the iterator has no elements.

Reply:
it's not necessarily less readable either and it's not critical so if you prefer the way you did i'm fine with it

Reply:
That's right, It only happens in the unhappy path, nevermind!